waffle-compiler/src/flattener.ts

```diff
@@ -0,0 +1,100 @@
+import {loadConfig} from './loadConfig';
+import {Config, InputConfig, inputToConfig} from './config';
+import {ImportsFsEngine, resolvers} from '@resolver-engine/imports-fs';
+import {gatherSources} from '@resolver-engine/imports';
+import {findInputs} from './findInputs';
+import {isWarningMessage} from './utils';
+import mkdirp from 'mkdirp';
+import fsx from 'fs-extra';
+import * as path from 'path';
+import {join} from 'path';
+
+export interface GatheredContractInterface {
+  url: string;
+  source: string;
+  provider: string;
+}
+
+export const IMPORT_SOLIDITY_REGEX = /^\s*import(\s+).*$/gm;
+
+export async function flattenProject(configPath?: string) {
+  await flattenAndSave(await loadConfig(configPath));
+}
+
+export async function flattenAndSave(input: InputConfig) {
+  const config = inputToConfig(input);
+  const output = await getContractDependency(config);
+  await processOutput(output, config);
+}
+
+async function getContractDependency(config: Config): Promise<GatheredContractInterface[][]> {
+  const resolver = ImportsFsEngine().addResolver(
+    resolvers.BacktrackFsResolver(config.nodeModulesDirectory)
+  );
+
+  const allContracts = findInputs(config.sourceDirectory);
+
+  const contractsDependency = await Promise.all(allContracts.map(async contract => {
+    return gatherSources(
+      [contract],
+      '.',
+      resolver
+    );
+  }));
+
+  return contractsDependency.map((contract: Array<GatheredContractInterface>) => {
+    return contract.map((dependency) => {
+      dependency.source = dependency.source.replace(IMPORT_SOLIDITY_REGEX, '');
+      return dependency;
+    });
+  });
```

Comment:
`contractsDependency` does not reflect the meaning of the variable. It's an array of contracts with all of their dependency files. Instead of trying to come up with a good name for it, I would combine these two iterations into one 😀